src/pure-js/deck.js

```diff
@@ -76,31 +80,27 @@ export default class DeckGL {
     this.needsRedraw = true;
     this.layerManager = null;
     this.effectManager = null;
-    autobind(this);
 
-    const {width, height, gl, glOptions, debug} = props;
+    // Bind methods
+    this._onRendererInitialized = this._onRendererInitialized.bind(this);
+    this._onRenderFrame = this._onRenderFrame.bind(this);
```

Comment:
Since we are removing autobind, shall we move towards using the array function style instead of these explicit bindings? will need transform plugin in the .babelrc though.

Reply:
> will need transform plugin in the .babelrc though.

Currently deck.gl uses pure ES6 and runs without transpilation in Node 6+ and modern browsers. If we start adding ES7+ features this will break.